prime-dotnet-webapi/Models/Statuses/SiteStatus.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Text.Json.Serialization;
+
+namespace Prime.Models
+{
+    [Table("SiteStatus")]
+    public class SiteStatus : BaseAuditable
+    {
+        [Key]
+        public int Id { get; set; }
+
+        public int SiteId { get; set; }
+
+        // [JsonIgnore]
+        // public Site Site { get; set; }
+
+        public SiteStatusType StatusType { get; set; }
+
+        public DateTimeOffset StatusDate { get; set; }
+
+
+        // TODO: Why not constructor?
```

Comment:
There are two reasons this was done like this for Enrollee statuses:
1. I generally prefer to use factory methods rather than constructors when properties are being set other than the given parameters. In this case, StatusDate is being set but isn't a parameter.
2. De-Serializing an object uses, by default, the empty constructor of the class. The curly brace object construction syntax also uses the empty constructor. This:
```C#
var object = new SiteStatus
{
    SiteId = siteId,
    StatusType = siteStatusType,
    StatusDate = DateTimeOffset.Now
}
```
is actually equivalent to
```C#
var object = new SiteStatus();
object.SiteId = siteId;
object.StatusType = siteStatusType;
object.StatusDate = DateTimeOffset.Now;
```

So we want the empty constructor on the class. If no constructor is specified, the empty one is added implicitly to the class. If *any* other constructor is specified, however, the implicit empty constructor is removed and you have to explicitly add it to the class if you want it. So you would have to add both the empty constructor and one that takes SiteStatusType and SiteId. In terms of lines of code, I went with 

```C#
public static SiteStatus FromType(SiteStatusType siteStatusType, int siteId)
{
    return new SiteStatus
    {
        SiteId = siteId,
        StatusType = siteStatusType,
        StatusDate = DateTimeOffset.Now
    };
}
```
vs
```C#
public SiteStatus() {}

public SiteStatus(SiteStatusType siteStatusType, int siteId)
{
    SiteId = siteId;
    StatusType = siteStatusType;
    StatusDate = DateTimeOffset.Now;
}
```

Reply:
Valid points